logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/sso/SSOInitiateLoginUseCase.kt

```diff
@@ -0,0 +1,50 @@
+package com.wire.kalium.logic.feature.auth.sso
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.configuration.ServerConfig
+import com.wire.kalium.logic.data.auth.login.SSOLoginRepository
+import com.wire.kalium.logic.feature.auth.ValidateUUIDUseCase
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.network.exceptions.KaliumException
+import io.ktor.http.HttpStatusCode
+
+sealed class SSOInitiateLoginResult {
+    data class Success(val requestUrl: String) : SSOInitiateLoginResult()
+
+    sealed class Failure : SSOInitiateLoginResult() {
+        object InvalidCode : Failure()
+        object InvalidRedirect : Failure()
+        class Generic(val genericFailure: CoreFailure) : Failure()
+    }
+}
+
+data class Redirects(val success: String, val error: String)
+
+interface SSOInitiateLoginUseCase {
+    suspend operator fun invoke(code: String, serverConfig: ServerConfig, redirects: Redirects? = null): SSOInitiateLoginResult
+}
+
+internal class SSOInitiateLoginUseCaseImpl(
+    private val ssoLoginRepository: SSOLoginRepository,
+    private val validateUUIDUseCase: ValidateUUIDUseCase
+) : SSOInitiateLoginUseCase {
+
+    override suspend fun invoke(code: String, serverConfig: ServerConfig, redirects: Redirects?): SSOInitiateLoginResult = suspending {
+        when {
+            !validateUUIDUseCase.invoke(code) -> return@suspending SSOInitiateLoginResult.Failure.InvalidCode
```

Comment:
This will not work since the sso code is `wire-{UUID}`, for now, it's fine to check the code validation before calling the Initiate endpoint, and assume it's valid until BE return otherwise.

Reply:
Yep, this is for the API request it takes the UUID part only, but the user input is descried here https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/18416054/Use+case+Entering+SSO+code#Code-format

Reply:
Ok so I changed it to `ValidateSSOCodeUseCase` which returns extracted uuid when valid